open_spiel/games/dark_hex.cc

```diff
@@ -0,0 +1,267 @@
+  black_view_.resize(board_size * board_size, CellState::kEmpty);
+  white_view_.resize(board_size * board_size, CellState::kEmpty);
+
+  kNumCells = board_size_ * board_size_;
+  kBitsPerAction = kNumCells + 1;
+  kLongestSequence = kNumCells * 2 - 1;
+}
+
+void DarkHexState::DoApplyAction(Action move) {
+  Player cur_player = CurrentPlayer(); //current player
+  auto& cur_view = (cur_player == Player(0) ? black_view_ : white_view_);
+
+  // Either occupied or not
+  if (state_.BoardAt(move) == CellState::kEmpty) {
+    state_.ApplyAction(move);
+  } else {}
+
+  SPIEL_CHECK_EQ(cur_view[move], CellState::kEmpty);
+  cur_view[move] = state_.BoardAt(move);
+  action_sequence_.push_back(std::pair<int,Action>(cur_player, move));
+}
+
+std::vector<Action> DarkHexState::LegalActions() const {
+  if (IsTerminal()) return {};
+  std::vector<Action> moves;
+  const Player player = CurrentPlayer();
+  const auto& cur_view = (player == Player{0} ? black_view_ : white_view_);
+
+  for (Action move = 0; move < kNumCells; ++move){
+    if (cur_view[move] == CellState::kEmpty){
+      moves.push_back(move);
+    }
+  }
+
+  return moves;
+}
+
+std::string DarkHexState::ViewToString(Player player) const {
+  const auto& cur_view = (player == Player{0} ? black_view_ : white_view_);
+  std::string str;
+
+  // TODO: ??
+  // Change here for row & cols after customizing for x, y
+  int num_rows = board_size_, num_cols = board_size_;
+  for (int r = 0; r < num_rows; ++r){
+    for (int c = 0; c < num_cols; ++c){
+      absl::StrAppend(&str, StateToString(cur_view[r * num_cols + c]));
+    }
+    if (r < (num_rows - 1)) absl::StrAppend(&str, "\n");
+  }
+  return str;
+}
+
+std::string DarkHexState::ActionSequenceToString(Player player) const {
+  SPIEL_CHECK_GE(player, 0);
+  SPIEL_CHECK_LT(player, num_players_);
+
+  std::string str;
+  for (const auto& player_with_action: action_sequence_) {
+    if (player_with_action.first == player) {
+      str.append(std::to_string(player_with_action.first));
+      str.push_back(',');
+      str.append(std::to_string(player_with_action.second));
+      str.push_back(' ');
+    } else if (obs_type_ == ObservationType::kRevealNumTurns) {
+      str.append(std::to_string(player_with_action.first));
+      str.append(",? ");
+    } else {
+      SPIEL_CHECK_EQ(obs_type_, ObservationType::kRevealNothing);
+    }
+  }
+  return str;
+}
+
+std::string DarkHexState::InformationStateString(Player player) const {
+  SPIEL_CHECK_GE(player, 0);
+  SPIEL_CHECK_LT(player, num_players_);
+  return ViewToString(player) + "\n"
+      + std::to_string(history_.size()) + "\n"
+      + ActionSequenceToString(player);
+}
+
+void DarkHexState::InformationStateTensor(Player player,
+                                          absl::Span<float> values) const {
+  SPIEL_CHECK_GE(player, 0);
+  SPIEL_CHECK_LT(player, num_players_);
+
+  const auto& player_view = (player == Player{0} ? black_view_ : white_view_);
```

Comment:
Player will always be an integer. No need to use Player{value}. 

Elsewhere too.